sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2050 @@
+        );
+        let file = File::create(&path)?;
+        Ok(Self {
+            temp_dir,
+            path: path.into(),
+            file,
+        })
+    }
+
+    /// Get the path of the temporary file
+    pub fn get_path(&self) -> &Path {
+        self.path.as_ref()
+    }
+
+    /// Get a mutable reference to the temporary file
+    pub fn get_mut_file(&mut self) -> &mut File {
+        &mut self.file
+    }
+}
+
+/// Supported extension and mime-types
+static SUPPORTED_TYPES: [&str; 4] = [
+    "application/font-woff",
+    "application/x-font-woff",
+    "font/woff",
+    "woff",
+];
+
+/// Four-character tag which names a font table.
+///
+/// TBD - Debug needs help,
+#[derive(Clone, Copy, Eq, Hash, Ord, PartialEq, PartialOrd)]
+struct TableTag {
+    data: [u8; 4],
+}
+
+impl TableTag {
+    fn as_bytes(&self) -> &[u8] {
+        &self.data
+    }
+    pub fn new<T: Read + Seek + ?Sized>(source_stream: &mut T) -> Result<Self> {
+        Ok(Self {
+            data: [source_stream.read_u8()?, source_stream.read_u8()?,  // Ick, YHGTBKM...
+                   source_stream.read_u8()?, source_stream.read_u8()?],
+        })
+    }
+}
+
+impl std::fmt::Display for TableTag {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}{}{}", self.data[0], self.data[1], self.data[2], self.data[3])
+    }
+}
+
+impl std::fmt::Debug for TableTag {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}{}{}", self.data[0], self.data[1], self.data[2], self.data[3])
+    }
+}
+
+/// Ad-hoc "tag" names for font-file entities which are not really tables,
+/// but which it is frequently convenient to treat as if they were. We re-use
+/// the font-file magic strings, because they are extremely unlikely to appear
+/// as table tags.
+const WOFF_HEADER_TAG: TableTag = TableTag { data: *b"wOFF", };
+const WOFF_METADATA_TAG: TableTag = TableTag { data: *b"wXML", };
+const WOFF_PRIVATE_DATA_TAG: TableTag = TableTag { data: *b"wPRV", };
+/// Pseudo-tag for the table directory.
+const TABLE_DIRECTORY_TAG: TableTag = TableTag { data: *b"\t\t\t\t", };
+/// Tag for the 'C2PA' table.
+const C2PA_TABLE_TAG: TableTag = TableTag { data: *b"C2PA", };
+/// Tag for the 'head' table in a font.
+const HEAD_TABLE_TAG: TableTag = TableTag { data: *b"head", };
+
+/// 32-bit font-format identifier.
+///
+/// Embedded OpenType and MicroType Express formats cannot be detected with a
+/// simple magic-number sniff. Conceivably, EOT could be dealt with as a
+/// variation on SFNT, but MTX needs more exotic handling.
+enum Magic {
+    /// 'OTTO' - OpenType
+    OpenType = 0x4f54544f,
+    /// FIXED 1.0 - TrueType (or possibly v1.0 Embedded OpenType)
+    TrueType = 0x00010000,
+    /// 'wOFF' - WOFF 1.0
+    Woff = 0x774f4646,
+    /// 'wOF2' - WOFF 2.0
+    Woff2 = 0x774f4632,
+}
+
+/// Used to attempt conversion from u32 to a Magic value.
+impl TryFrom<u32> for Magic {
+    type Error = ();
```

Comment:
I would say use the crates error for the type here and then on line 376 return an error that would help identify the real problem.

Reply:
Looking into it.